### PR TITLE
Don't serialize empty GUID to null

### PR DESCRIPTION
**Pull request by crobibero**
Fixes https://github.com/jellyfin/jellyfin/issues/4029#issuecomment-740209476

Note: this still fixes https://github.com/jellyfin/jellyfin/issues/4566

### CODE REVIEW

File: MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs [hunk 1]
```diff
@@ -13,21 +13,13 @@ public class JsonGuidConverter : JsonConverter<Guid>
...
         {
-            if (value == Guid.Empty)
-            {
-                writer.WriteNullValue();
-            }
-            else
-            {
-                writer.WriteStringValue(value);
-            }
+            writer.WriteStringValue(value);
```
Comment (BaronGreenback) [inline MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs:22]: Will this stop all the 0000000-000-000 guids?

**Review by BaronGreenback — APPROVED** [review]

File: MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs (hunk 1 above)
Comment (crobibero) [reply MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs:22]: This writes the Guid.Empty values, if we don't want a Guid to be sent it should be marked as nullable, and the NullableStructConverter will handle it

Comment (BaronGreenback) [reply MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs:22]: See what you've done now.

File: MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs [hunk 2]
```diff
@@ -13,21 +13,13 @@ public class JsonGuidConverter : JsonConverter<Guid>
...
-
             return guidStr == null ? Guid.Empty : new Guid(guidStr);
         }
 
         /// <inheritdoc />
         public override void Write(Utf8JsonWriter writer, Guid value, JsonSerializerOptions options)
         {
-            if (value == Guid.Empty)
-            {
-                writer.WriteNullValue();
```
Comment (cvium) [inline MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs:25]: Why don't we want this anymore?

**Comment by nielsvanvelzen** [comment]
BaseItemDto is not the only place using uuid's. Do all other models use the 0000-000-000 uuid?

File: MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs (hunk 2 above)
Comment (crobibero) [reply MediaBrowser.Common/Json/Converters/JsonGuidConverter.cs:25]: This causes issues with the generated OpenApi spec, where the GUID in the spec is not-null, but null is returned.

**Comment by crobibero** [comment]
> 
> 
> BaseItemDto is not the only place using uuid's. Do all other models use the 0000-000-000 uuid?

The uuid is defaulted to 0000 if not set, so probably

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**